airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java

```diff
@@ -93,6 +116,7 @@ public AutoCloseableIterator<JsonNode> queryTableIncremental(JdbcDatabase databa
                                                                JDBCType cursorFieldType,
                                                                String cursor) {
     LOGGER.info("Queueing query for table: {}", tableName);
+    String limit = !this.limit_per_read.equals("0") ? String.format("TOP %s", this.limit_per_read) : "";
```

Comment:
This logic of setting to empty string if 0 or empty otherwise setting to TOP _ could get refactored into some method to DRY up